# Analiza cen wynajmu mieszkań w Szczecinie

## Celem projektu jest prosta analiza ofert wynajmu mieszkań w Szczecinie na podstawie danych z okresu listopad 2023 - czerwiec 2024.

## Import wymaganych bibliotek

In [1]:
import pandas as pd


## Przygotowanie listy plików z danymi

In [2]:
dane = [
    ("Dane/apartments_rent_pl_2023_11.csv", "2023-11"),
    ("Dane/apartments_rent_pl_2023_12.csv", "2023-12"),
    ("Dane/apartments_rent_pl_2024_01.csv", "2024-01"),
    ("Dane/apartments_rent_pl_2024_02.csv", "2024-02"),
    ("Dane/apartments_rent_pl_2024_03.csv", "2024-03"),
    ("Dane/apartments_rent_pl_2024_04.csv", "2024-04"),
    ("Dane/apartments_rent_pl_2024_05.csv", "2024-05"),
    ("Dane/apartments_rent_pl_2024_06.csv", "2024-06")
]

## Wczytanie i połączenie danych

In [5]:
lista_danych = []

for plik, miesiac in dane:
    dane_miesiac = pd.read_csv(plik)
    dane_miesiac["month"] = miesiac
    lista_danych.append(dane_miesiac)

df = pd.concat(lista_danych, ignore_index=True)

df.head(10)

,id,city,type,squareMeters,rooms,floor,floorCount,buildYear,latitude,longitude,...,ownership,buildingMaterial,condition,hasParkingSpace,hasBalcony,hasElevator,hasSecurity,hasStorageRoom,price,month
0,2a1a6db97ff122d6bc148abb6f0e498a,szczecin,blockOfFlats,52.00,2.0,3.0,3.0,2008.0,53.460535,14.545416,...,condominium,brick,NaN,yes,yes,no,no,no,3500,2023-11
1,5a0cb42c088eadf75aa98a4118640048,szczecin,blockOfFlats,57.00,3.0,NaN,10.0,1976.0,53.440955,14.566024,...,condominium,NaN,premium,no,yes,yes,no,no,3000,2023-11
2,996722490efcb4b34d48bb84a9a72838,szczecin,NaN,43.40,2.0,3.0,4.0,NaN,53.435746,14.560764,...,condominium,NaN,NaN,no,yes,no,no,no,1900,2023-11
3,50d75c688530b9a61ba505301e0b94a9,szczecin,apartmentBuilding,32.60,2.0,5.0,6.0,2021.0,53.421582,14.541572,...,condominium,brick,premium,no,yes,yes,no,no,2500,2023-11
4,9b187f4203adabf466fd08610d8f4e5e,szczecin,apartmentBuilding,72.00,3.0,6.0,6.0,2014.0,53.450010,14.547230,...,condominium,brick,premium,no,yes,yes,no,yes,2999,2023-11
5,ba4d70d65d1024ee13929ca82bfc92f6,szczecin,blockOfFlats,52.00,2.0,1.0,3.0,2006.0,53.427400,14.488100,...,condominium,concreteSlab,NaN,yes,no,no,no,no,2300,2023-11
6,195bf5124c9fd720ca3ff7aeff9ca83f,szczecin,blockOfFlats,45.94,2.0,4.0,10.0,1978.0,53.433567,14.539891,...,condominium,concreteSlab,NaN,no,yes,yes,no,no,2500,2023-11
7,03a41e8ec8ebdc1dd5a86d65ae3de54e,szczecin,blockOfFlats,54.00,2.0,NaN,3.0,2002.0,53.412600,14.541100,...,condominium,brick,premium,yes,yes,yes,no,yes,2600,2023-11
8,aae4994ab375e107dcc375f3360bd3d9,szczecin,tenement,32.00,1.0,2.0,4.0,1930.0,53.434836,14.548864,...,condominium,brick,premium,no,no,NaN,no,no,1500,2023-11
9,b83458393eb79b3ecdb13de56a6e2676,szczecin,blockOfFlats,76.38,3.0,1.0,1.0,1954.0,53.435478,14.512039,...,condominium,brick,premium,yes,no,no,no,no,3600,2023-11


## Sprawdzenie rozmiaru danych, liczby wierszy i kolumn

In [6]:
df.shape

(70847, 29)

# Podstawowe informacje o danych

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 70847 entries, 0 to 70846
Data columns (total 29 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   id                    70847 non-null  object 
 1   city                  70847 non-null  object 
 2   type                  54085 non-null  object 
 3   squareMeters          70847 non-null  float64
 4   rooms                 70847 non-null  float64
 5   floor                 62234 non-null  float64
 6   floorCount            69468 non-null  float64
 7   buildYear             51165 non-null  float64
 8   latitude              70847 non-null  float64
 9   longitude             70847 non-null  float64
 10  centreDistance        70847 non-null  float64
 11  poiCount              70847 non-null  float64
 12  schoolDistance        70832 non-null  float64
 13  clinicDistance        70797 non-null  float64
 14  postOfficeDistance    70821 non-null  float64
 15  kindergartenDistanc

Dane mają 70847 wierszy i 29 kolumn. Na tym etapie widać, że część kolumn posiada braki danych, np. `type`, `floor` i inne. Braki zostaną dokładniej sprawdzone w dalszej części projektu.

## Sprawdzenie nazw kolumn

In [9]:
df.columns

Index(['id', 'city', 'type', 'squareMeters', 'rooms', 'floor', 'floorCount',
       'buildYear', 'latitude', 'longitude', 'centreDistance', 'poiCount',
       'schoolDistance', 'clinicDistance', 'postOfficeDistance',
       'kindergartenDistance', 'restaurantDistance', 'collegeDistance',
       'pharmacyDistance', 'ownership', 'buildingMaterial', 'condition',
       'hasParkingSpace', 'hasBalcony', 'hasElevator', 'hasSecurity',
       'hasStorageRoom', 'price', 'month'],
      dtype='object')

W danych znajdują się informacje między innymi o mieście, cenie, metrażu, liczbie pokoi, piętrze, roku budowy, odległościach od różnych punktów oraz dodatkowych cechach mieszkania.

## Sprawdzenie dostępnych miast

In [10]:
df["city"].value_counts()

city
warszawa       28414
krakow         12661
wroclaw         6676
lodz            5214
poznan          3890
gdansk          3595
katowice        2669
gdynia          1483
szczecin        1482
bydgoszcz       1478
lublin          1313
czestochowa      661
rzeszow          614
bialystok        440
radom            257
Name: count, dtype: int64

Najwięcej ofert znajduje się w Warszawie oraz Krakowie, najmniej w Częstochowie, Rzeszowie, Białymstoku i Radomiu.

W danych znajduje się wiele miast, do analizy wybrany został Szczecin.